bin/minifollowups/pycbc_injection_minifollowup

```diff
@@ -14,20 +14,96 @@
 from pycbc.events import select_segments_by_definer, coinc
 from pycbc.results import layout
 from pycbc.detector import Detector
-import pycbc.workflow.minifollowups as mini
 from pycbc.workflow.core import resolve_url_to_file
-import pycbc.version, pycbc.pnutils
 from pycbc.io.hdf import SingleDetTriggers, HFile
+from pycbc.conversions import mchirp_from_mass1_mass2
+
 
-parser = argparse.ArgumentParser(description=__doc__[1:])
+def sort_injections(args, inj_group, missed, sort_by):
+    """Return an array of indices to sort the missed injections according to a
+    metric of choice.
+
+    Parameters
+    ----------
+    args : object
+        CLI arguments parsed by argparse.
+    inj_group : h5py group object
+        HDF5 group object containing the injection definition.
+    missed : array
+        Array of indices of missed injections into `inj_group`.
+    sort_by : str
+        Which metric to sort by.
+
+    Returns
+    -------
+    sorting : array
+        Array of indices that will sort the missed injections as requested.
+    """
+    if 'optimal_snr' in sort_by:
+        optimal_snrs = [
+            inj_group[dsn][:][missed] for dsn in inj_group.keys()
+            if dsn.startswith('optimal_snr_')
+        ]
+        assert optimal_snrs, 'These injections do not have optimal SNRs'
+
+    if sort_by == 'decisive_optimal_snr':
+        # descending order of decisive (2nd largest) optimal SNR
+        dec_snr = numpy.array([
+            sorted(snrs)[-2] for snrs in zip(*optimal_snrs)
+        ])
+        if args.maximum_decisive_snr is not None:
+            # By setting to 0, these injections will not be considered
+            dec_snr[dec_snr > args.maximum_decisive_snr] = 0
+        return dec_snr.argsort()[::-1]
+
+    if sort_by == 'network_optimal_snr':
+        # descending order of network optimal SNR
+        optimal_snrs = numpy.vstack(optimal_snrs)
+        net_opt_snrs_squared = (optimal_snrs ** 2).sum(axis=0)
+        return net_opt_snrs_squared.argsort()[::-1]
+
+    if sort_by == 'decisive_eff_dist':
+        # ascending order of decisive (2nd smallest) chirp distance
+        eff_dists = []
+        for ifo in args.single_detector_triggers:
+            eff_dist = Detector(ifo).effective_distance(
+                inj_group['distance'][:][missed],
+                inj_group['ra'][:][missed],
+                inj_group['dec'][:][missed],
+                inj_group['polarization'][:][missed],
+                inj_group['tc'][:][missed],
+                inj_group['inclination'][:][missed]
+            )
+            eff_dists.append(eff_dist)
+        dec_eff_dist = sorted(eff_dists)[-2]
+        mchirp = mchirp_from_mass1_mass2(
+            inj_group['mass1'][:][missed],
+            inj_group['mass2'][:][missed]
+        )
+        dec_chirp_dist = pycbc.pnutils.chirp_distance(dec_dist, mchirp)
+        return dec_chirp_dist.argsort()
+
+    raise ValueError(f'Invalid injection sorting method "{sort_by}"')
```

Comment:
On the principle 'error as soon as possible' I would prefer to define a list of allowed sort-by options outside this function and check the sort by argument is in the list before doing anything else.  Then you can also print out the list in the error message ('Allowed methods are ...')

Reply:
supply as choices in the argparse then?

Reply:
This has not been addressed

Reply:
Fair enough, this requires some duplication with the present organization of the code, but we can straighten that in that refactor PR I was talking about earlier.